transcom/mymove
# B 20942 int

## [B-20942](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20942)

> [!WARNING]  
> Commit 2b95b2f is made directly on the int branch to resolve merge conflicts between other SSW features in the server tests only. This solution to the merge conflicts should be made the same way when moving the feature to main. (The force push was just to resolve divergent branches between local and git when updating off of integrationTesting). Additionally, please take note of the merge conflict resolutions for potential future main conflicts.

## Summary
This PR adjusts the SSW PDF Template to change the "Actual Move Date" field to instead read "Pick-up date". It also additionally ensures that for a Payment Packet, the actual pick-up date (actual move date is the same value) is used. But, for an AOA packet, the expected departure date is used instead.

Due to git not being able to merge PDF files, this will have to be downstream of the latest template change, [B-20939](https://github.com/transcom/mymove/pull/13538)

Further, some code was refactored to remove an erroneous loop that was running for only one ppm shipment in a FormatAllShipments function. This was instead combined with the formatCurrentShipment function, and renamed formatShipment. These changes make things more clear, and it helps clean things up for an upcoming feature supporting multiple shipment information in a single SSW.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the app as a service member.
2. Create a move.
3. Create a PPM Shipment with an advance, take note of your expected departure date.
4. Submit the shipment.
5. Log in as a service counselor. 
6. Approve the orders, submit the details, download the AOA Packet.
7. Verify that the date field reads "Pick-up date", with the expected departure date.
8. Log back in as service member. Begin the closeout process, take note of your actual move date.
9. Complete and submit the closeout process.
10. Log back in as a service counselor.
11. Close out the PPM. Make sure the actual move date matches step 8.
12. Once submitted, download the payment packet.
13. Ensure the Pick-up date matches steps 8 and 11.

### Backend


## Screenshots
<img width="565" alt="image" src="https://github.com/user-attachments/assets/4ef87e2c-0864-46ee-a85f-a04469ca969c">
